packages/cli/src/lib/setup/setupUsers.ts

```diff
@@ -1,129 +1,165 @@
-    }
-    if (!options.processExit) {
-        throw new Error('Invalid arguments: processExit is missing');
+        this.objects = options.objects;
+        this.processExit = options.processExit;
     }
 
-    const objects = options.objects;
-    const processExit = options.processExit;
-    const { EXIT_CODES } = require('@iobroker/js-controller-common');
-
-    this.addUser = function (user, pw, callback) {
+    /**
+     * Adds new user to system
+     *
+     * @param user username
+     * @param pw password
+     * @param callback
+     */
+    addUser(user: string, pw: string, callback: ioBroker.ErrorCallback): void {
         // user id's should be case insensitive
         const _user = user.replace(/\s/g, '_').toLowerCase();
-        objects.getObject('system.user.' + _user, (err, obj) => {
+        this.objects.getObject(`system.user.${_user}`, (err, obj) => {
             if (obj) {
                 return tools.maybeCallbackWithError(callback, 'User yet exists');
             } else {
-                objects.setObject(
-                    'system.user.' + _user,
+                this.objects.setObject(
+                    `system.user.${_user}`,
                     {
                         type: 'user',
                         common: {
                             name: user,
-                            enabled: true
+                            enabled: true,
+                            password: ''
                         },
-                        from: 'system.host.' + tools.getHostName() + '.cli',
+                        from: `system.host.${tools.getHostName()}.cli`,
                         ts: Date.now(),
                         native: {}
                     },
                     err => {
                         if (!err) {
-                            that.setPassword(user, pw, callback);
+                            this.setPassword(user, pw, callback);
                         } else {
                             return tools.maybeCallbackWithError(callback, err);
                         }
                     }
                 );
             }
         });
-    };
+    }
 
-    this.isUser = function (user, callback) {
+    /**
+     * Checks if user exists
+     *
+     * @param user username
+     */
+    async isUser(user: string): Promise<boolean> {
         const _user = user.replace(/\s/g, '_').toLowerCase();
-        objects.getObject(`system.user.${_user}`, (err, obj) => {
-            return tools.maybeCallbackWithError(callback, null, !!obj);
-        });
-    };
+        let isExisting = false;
+        try {
+            isExisting = await this.objects.objectExists(`system.user.${_user}`);
+        } catch {
+            // ignore
+        }
+
+        return isExisting;
+    }
 
-    this.setPassword = function (user, pw, callback) {
+    /**
+     * Set password for specific user
+     *
+     * @param user username
```

Comment:
Are you planning to add useful documentation for those params? Else this is pretty much:
![grafik](https://user-images.githubusercontent.com/17641229/216792490-4b99d18c-0adb-42b0-81a5-4d71a64a275f.png)

Reply:
For me it is enough to know it’s just a username not an id or some special format.

Reply:
let's just call the parameter `username` then?